datafusion/physical-optimizer/src/test_utils.rs

```diff
@@ -207,6 +251,56 @@ pub fn aggregate_exec(input: Arc<dyn ExecutionPlan>) -> Arc<dyn ExecutionPlan> {
     )
 }
 
+pub fn aggregate_exec_monotonic(input: Arc<dyn ExecutionPlan>) -> Arc<dyn ExecutionPlan> {
+    let schema = input.schema();
+    let aggregate_expr =
+        vec![
+            AggregateExprBuilder::new(count_udaf(), vec![col("d", &schema).unwrap()])
+                .schema(Arc::clone(&schema))
+                .alias("count")
+                .build()
+                .map(Arc::new)
+                .unwrap(),
+        ];
+    Arc::new(
+        AggregateExec::try_new(
+            AggregateMode::Final,
```

Comment:
```suggestion
            AggregateMode::Single,
```

Final is used in 2-step aggregation, should be Single here